accepted/2024/unified-vs-settings.md

```diff
@@ -0,0 +1,160 @@
+
+#### Error messages
+
+When the `NuGet.Config` has an invalid value, the UI will show a blank selection, or unchecked checkbox, and will be accompanied by a simple error message indicating that the value in the backing store was somehow invalid.
+
+Any action taken inside the USX page will overwrite that invalid value in the backing store (`NuGet.Config`) according to `NuGet.Configuration`.
+
+For any fatal errors, such as unable to load the `NuGet.Config` at all due to parsing errors, or similar, then an error message will be shown under the "General" section label.
+
+#### Help icons
+
+Each UI element in USX can have a help link attached.
+In the mockup, these are shown as a circled `?` icon.
+Pressing one of these opens the relevant Documentation page, and is anchored to the corresponding section in the document.
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+### Registration
+
+The NuGet.Client repository will maintain a `registration.json` file to declare all VS Settings, including localizible names, data types, default values, command implementations, and any dependencies among the values of other settings.
+
+The `registration.json` will use the `"type": "external"` mode because our product stores configuration in a custom backing store: One or more `NuGet.Config` files.
+
+### Command: Clear Local Resources
+
+NuGet.Client will create a new modal dialog will be developed to indicate that the Clear Local Resources command is in-progress and has completed.
+The modal dialog will replace our WinForms controls that indicate the same information.
+
+Since USX doesn't support dynamically changing text blocks, the modal is the simplest and most effective form of UI we can adopt, while also allowing us to expand the capabilities of this command in VS in the future.
+
+The invocation of the `LocalsCommandRunner` which is supported by legacy VS settings will be adapted to a USX Command.
+A USX Command invokes a command in VS using the VS SDK and the extensible menu command system.
+Therefore, our Visual Studio Command Table (VSCT) file will be updated with a new command, which executes `LocalsCommandRunner` in the following way:
+
+1. Open a modal dialog window.
+    - State that NuGet resources are being cleared.
+    - Prevent the user from accessing VS components like the Restore or the NuGet Package Manager since this could cause conflicts with the operation of clearing local resources.
+    - Show an indeterminate progress indicator.
+    - A Cancel button cannot be supported at this time.
+    (See Future Possibilities)
+1. Execute `LocalsCommandRunner` in the same manner as legacy VS Settings.
+1. Display text indicating the completion of the command (or any error messages) and a close button on the modal dialog.
```

Comment:
Maybe the dialog should just close automatically when complete?  Just an idea, it's your call.

Reply:
I think I'll leave it for now as I'd like to show the completion information. Easy to change later, if necessary.